explore/src/main/scala/explore/programs/ProgramsPopup.scala

```diff
@@ -124,7 +127,7 @@ object ProgramsPopup:
       val programInfoViewListOpt: Option[List[View[ProgramInfo]]] =
         programInfosViewOpt.map:
           _.toListOfViews
-            .map(_._2.withOnMod(pi => newProgramId.set(none) >> doSelectProgram(pi.id)))
```

Comment:
Unrelated bug fix. Whenever any program was edited on the dialog, new or old, including the name or existence status, the program was immediately selected and the popup went away. Among other oddities, this would select deleted programs. I moved the `select program` bit to where the name is edited, and only select the program when the name is saved if the program was just created.